Orchestration/IntegrationService/Implementations/AzureResourceManagerDeploymentService.ps1

```diff
@@ -697,4 +697,52 @@ Class AzureResourceManagerDeploymentService: IDeploymentService {
             throw $_;
         }
     }
+
+    [void] RemoveResourceGroupLock([guid] $subscriptionId,
+                                   [string] $resourceGroupName) {
+        try {
+            $scope = "/subscriptions/$subscriptionId/resourceGroups/$resourceGroupName"
+            $allLocks = Get-AzResourceLock -Scope $scope -ErrorAction SilentlyContinue | Where-Object "ProvisioningState" -ne "Deleting"
```

Comment:
I am just trying to be safe, otherwise we need to assume that you did a Set-AzContext on the right subscription, or I have to do it in the code and we might get into the same blocking issue we have by invoking Set-AzContext multiple times.